### ISSUE TITLE
performance disagreement in graph convolutional networks example notebook

### ISSUE THREAD

**Issue by davidwhealey** [deepchem/deepchem#950]
in the graph_convolutional_networks_for_tox21 example notebook, the model is first trained using GraphConvTensorGraph, and then the whole thing is built up somewhat from scratch again to show how it works, concluding "The model we've constructed behaves nearly identically to `GraphConvTensorGraph`.

Running the notebook, however, gives mean AUC of ~.7-.8 with the GraphConvTensorGraph, but 0.5 with the scratch model that is intended to replicate it.

**Comment by rbharath** [comment]
You'll need to train the scratch model longer. Here's the relevant text from the notebook:

```
# Epochs set to 1 to render tutorials online.
# Set epochs=10 for better results.
tg.fit_generator(data_generator(train_dataset, epochs=1))
```

**Comment by davidwhealey** [comment]
should have mentioned that they are trained with the same number of epochs (10 in both cases)

**Comment by Dgelemi** [comment]
I can reproduce the results. The model building steps look to be the same though. Both took 90s on my machine and very similar Average loss after 10 epocs (384 vs 378).
Could it be that the  tg.evaluate_generator lines are wrong?

**Comment by davidwhealey** [comment]
@Dgelemi by "I can reproduce the results" do you mean you can reproduce the bug, or that your scratch model reproduces the GraphConvTensorGraph model's results?

**Comment by Dgelemi** [comment]
@davidwhealey , sorry, I mean that I reproduce your results, with the model build from scratch giving random AUC.

**Comment by lilleswing** [comment]
I just started looking at this --

The scratch model reproduces the GraphConvTensorGraph results.  There is a bug in the evaluation and collection of the metrics as @Dgelemi suggested.

There is also a newer bug introduced by https://github.com/deepchem/deepchem/pull/949/files on the model creation itself.

It's late on the east coast but I can fix the evaluation and collection of metrics code tomorrow.

**Comment by rbharath** [comment]
Closing since I think #954 solved this issue.
